static/js/util/util.js

```diff
@@ -575,6 +578,18 @@ export function sortedCourseRuns(program: Program): Array<CourseRun> {
   )
 }
 
+export function programBackendName(program: Program): string {
+  const courses = program.courses
+  if (
+    R.any(
+      R.equals(COURSEWARE_BACKEND_MITXONLINE),
+      R.pluck("courseware_backend", R.unnest(R.pluck("runs", courses)))
+    )
+  ) {
+    return COURSEWARE_BACKEND_NAMES[COURSEWARE_BACKEND_MITXONLINE]
+  }
+  return COURSEWARE_BACKEND_NAMES[COURSEWARE_BACKEND_EDXORG]
```

Comment:
I was looking around and i found a place where we already have an added property as for checking if a program is MITxOnline based where we already have a logic to calculate, I think we can reuse that to keep it in sync everywhere. So what it will be like is:

1. After [this line](https://github.com/mitodl/micromasters/blob/master/dashboard/api.py#L196) we add a property like: 
```
"has_mitxonline_courses": mmtrack.program.has_mitxonline_courses
```

2. After [this line](https://github.com/mitodl/micromasters/blob/master/static/js/flow/programTypes.js#L43) we add :
```
has_mitxonline_courses:     boolean
```

3. Our fuction `programBackendName` will change something like:
```
export function programBackendName(program: Program): string {
  if (program.has_mitxonline_courses) {
    return COURSEWARE_BACKEND_NAMES[COURSEWARE_BACKEND_MITXONLINE]
  }
  else {
    return COURSEWARE_BACKEND_NAMES[COURSEWARE_BACKEND_EDXORG]
  }
}
```